### PR TITLE
drivers/sx127x: Fix misc variable types and lengths

### PR DESCRIPTION
**Pull request by kYc0o**
While compiling tests/driver_sx127x with clang in OS X it discovers several minor issues.

This PR fixes them.

### CODE REVIEW

File: drivers/include/sx127x.h [hunk 1]
```diff
@@ -208,7 +208,7 @@ enum {
  */
 typedef struct {
     uint16_t preamble_len;             /**< Length of preamble header */
-    uint8_t power;                     /**< Signal power */
+    int8_t power;                      /**< Signal power */
```
Comment (aabadie) [inline drivers/include/sx127x.h]: This change doesn't seem to be related to compiler issues but it's a valid one regarding LoRa: the power in dBm can indeed be negative.

Note that you also need to update the netdev _set function according to this change.

Comment (kYc0o) [reply drivers/include/sx127x.h]: clang rise an error of assigning negative values to this variable. Besides the fact that yes, conceptually this should be a `uint8_t`.

The netdev change was already done, that's why Murdock agrees.

Comment (aabadie) [reply drivers/include/sx127x.h]: I think you misunderstood my comment. Using an `int8_t` is correct since power can be negative. But the netdev _set function should be adapted as well and this is not done by this PR. See [this line](https://github.com/RIOT-OS/RIOT/pull/7851/files#diff-37ee9c4f2a20ec7a38d429c04255744aR445)
The line just is also erroneous as it returns `sizeof(uint16_t)` and it should be `sizeof(int8_t` according to this PR.

Comment (kYc0o) [reply drivers/include/sx127x.h]: Ok got it. Fixed.

**Comment by kYc0o** [comment]
Go!
